File: app/src/main/java/com/jacob/fruitoftek/kotlinschoolmanagement/model/Student.kt
package com.jacob.fruitoftek.kotlinschoolmanagement.model

import androidx.room.Entity
import androidx.room.PrimaryKey

@Entity(tableName = "students")
data class Student(
    @PrimaryKey val id: Int,
    val user_id: String,
    val first_name: String,
    val last_name: String,
    val gender: String,
    val dob: String?,
    val nationality: String?,
    val language_preferences: String?,
    val photo: String?,
    val cert_id: String?,
    val email: String?,
    val student_phone: String?,
    val password: String?,
    val admission_date: String?,
    val current_class: String?,
    val previous_school: String?,
    val document_name: String?,
    val cert_photo: String?,
    val health_conditions: String,
    val on_medications: String,
    val blood_type: String,
    val take_immunization: String,
    val dietary_restrictions: String,
    val restricted_diets: String?,
    val has_disability: String,
    val disability_type: String?,
    val fathers_name: String?,
    val fathers_occupation: String?,
    val fathers_contact: String?,
    val mothers_name: String?,
    val mothers_occupation: String?,
    val mothers_contact: String?,
    val guardians_name: String?,
    val guardians_occupation: String?,
    val guardians_contact: String?,
    val on_create: String?,
    val on_update: String?
)

File: php_endpoint/Student.php
<?php
class Student {
    public $id;
    public $user_id;
    public $first_name;
    public $last_name;
    public $gender;
    public $dob;
    public $nationality;
    public $language_preferences;
    public $photo;
    public $cert_id;
    public $email;
    public $student_phone;
    public $password;
    public $admission_date;
    public $current_class;
    public $previous_school;
    public $document_name;
    public $cert_photo;
    public $health_conditions;
    public $on_medications;
    public $blood_type;
    public $take_immunization;
    public $dietary_restrictions;
    public $restricted_diets;
    public $has_disability;
    public $disability_type;
    public $fathers_name;
    public $fathers_occupation;
    public $fathers_contact;
    public $mothers_name;
    public $mothers_occupation;
    public $mothers_contact;
    public $guardians_name;
    public $guardians_occupation;
    public $guardians_contact;
    public $on_create;
    public $on_update;

    private $pdo;

    public function __construct(PDO $pdo) {
        $this->pdo = $pdo;
    }

    public function getAll(): array {
        $stmt = $this->pdo->query("SELECT * FROM students");
        return $stmt->fetchAll(PDO::FETCH_ASSOC);
    }

    public function upsert(array $data) {
        $columns = array_keys($data);
        $placeholders = array_map(fn($c) => ":$c", $columns);

        $updates = [];
        foreach ($columns as $column) {
            if ($column !== 'id') {
                $updates[] = "$column = VALUES($column)";
            }
        }

        $sql = sprintf(
            'INSERT INTO students (%s) VALUES (%s) ON DUPLICATE KEY UPDATE %s',
            implode(',', $columns),
            implode(',', $placeholders),
            implode(',', $updates)
        );

        $stmt = $this->pdo->prepare($sql);
        foreach ($data as $key => $value) {
            $stmt->bindValue(":$key", $value);
        }
        $stmt->execute();
        return $this->pdo->lastInsertId();
    }
}